### PR TITLE
Support caffe

### PR DESCRIPTION
**Pull request by judgeeeeee**
### Summary:

support caffe

### Test Plan:

```release-note
NONE
```

### CODE REVIEW

**Comment by simon-cj** [comment]
Today I will learn some background knowledge, @gaocegege review first, ignore me

File: source/python/neuropod/backends/caffe/executor.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    def __init__(self, neuropod_path):
+        """
+        Load a Caffe neuropod
+
+        :param  neuropod_path:      The path to a Caffe neuropod package
+
+        """
+        super(CaffeNeuropodExecutor, self).__init__(neuropod_path)
+        # Create a folder to store the model
+        neuropod_data_path = os.path.join(neuropod_path, "0", "data")
```
Comment (gaocegege) [inline source/python/neuropod/backends/caffe/executor.py:38]: Why do we need 0 here?

File: source/python/neuropod/backends/caffe/packager.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import json
+import shutil
+import caffe
```
Comment (gaocegege) [inline source/python/neuropod/backends/caffe/packager.py:18]: Will we require installing caffe if using it?

File: source/python/neuropod/backends/caffe/executor.py (hunk 1 above)
Comment (judgeeeeee) [reply source/python/neuropod/backends/caffe/executor.py:38]: Follow the previous (other framework ) code style

File: source/python/neuropod/backends/caffe/packager.py (hunk 2 above)
Comment (judgeeeeee) [reply source/python/neuropod/backends/caffe/packager.py:18]: Yes, we just do something like wrapper

**Pull request merged**